aws/src/main/java/org/apache/iceberg/aws/s3/S3InputFile.java

```diff
@@ -20,12 +20,16 @@
 package org.apache.iceberg.aws.s3;
 
 import org.apache.iceberg.aws.AwsProperties;
+import org.apache.iceberg.encryption.NativeFileCryptoParameters;
+import org.apache.iceberg.encryption.NativelyEncryptedFile;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.SeekableInputStream;
 import org.apache.iceberg.metrics.MetricsContext;
 import software.amazon.awssdk.services.s3.S3Client;
 
-public class S3InputFile extends BaseS3File implements InputFile {
+public class S3InputFile extends BaseS3File implements InputFile, NativelyEncryptedFile {
+  private NativeFileCryptoParameters nativeDecryptionParameters;
```

Comment:
can you make the the parameter name match the class name?

Reply:
here, we can use a more focused naming. `NativeFileCryptoParameters` are per-file, and are symmetric (applicable to both encryption and decryption). In an `InputFile` object, we can drop the 'file' part, and narrow this down to decryption-only. So the `nativeDecryptionParameters` name might be more streamlined and informative than a `nativeFileCryptoParameters`.
But if Iceberg has a convention of matching the filed names to the class names, then sure, I'll make this change.